pkgs/flox/default.nix

```diff
@@ -40,6 +41,7 @@
       NIX_BIN = "${flox-bash}/libexec/flox/nix";
       GIT_BIN = "${gitMinimal}/bin/git";
       PARSER_UTIL_BIN = parser-util.outPath + "/bin/parser-util";
+      PKGDB_BIN = flox-pkgdb.outPath + "/bin/pkgdb";
```

Comment:
Seems something is wrong on macOS:
```
> flox build .#flox
> FLOX_FEATURES_CHANNELS=rust ./result/bin/flox search hello
ERROR: pkgdb exited with status code -1:
    dyld[67950]: Library not loaded: lib/libflox-pkgdb.dylib
      Referenced from: <no uuid> /nix/store/yraivsr242555nlzkvzhlzn94lr0m6na-flox-pkgdb-0.1.0/bin/pkgdb
      Reason: tried: 'lib/libflox-pkgdb.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlib/libflox-pkgdb.dylib' (no such file), 'lib/libflox-pkgdb.dylib' (no such file), '/usr/local/lib/libflox-pkgdb.dylib' (no such file), '/usr/lib/libflox-pkgdb.dylib' (no such file, not in dyld cache)
```

Reply:
Good catch I think I botched the `-Wl,-rpath` flag for Darwin.
I'll get a patch sent over with a new GHA test to cover "running the installed form of the package".

Reply:
This fixes the issue: https://github.com/flox/pkgdb/pull/55
Once that merges I can bump the lockfile.

Reply:
should be fixed now

Reply:
Hmm that definitely fixed something over in pkgdb:
Before
```
> ./result/bin/pkgdb --help
dyld[48587]: Symbol not found: __ZN4flox16InputPreferences5clearEv
  Referenced from: <no uuid> /nix/store/yraivsr242555nlzkvzhlzn94lr0m6na-flox-pkgdb-0.1.0/bin/pkgdb
  Expected in:     <no uuid> /Users/matthew/pkgdb/lib/libflox-pkgdb.dylib
zsh: abort      ./result/bin/pkgdb --help
```
After
```
> ./result/bin/pkgdb --help
Usage: pkgdb [--help] [--quiet] [--verbose] {get,resolve,scrape,search}
...
```

But flox search is still failing:
```
> git rev-parse HEAD
86e1974b506d87ab9eaeadfe91f4208b20d950b3
> flox build .#flox
> FLOX_FEATURES_CHANNELS=rust ./result/bin/flox search hello
ERROR: pkgdb exited with status code -1:
    dyld[48715]: Library not loaded: lib/libflox-pkgdb.dylib
      Referenced from: <no uuid> /nix/store/769dik0r8l7p43cwvc11rcfsnpfwyzwq-flox-pkgdb-0.1.0/lib/libflox-pkgdb.dylib
      Reason: tried: 'lib/libflox-pkgdb.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlib/libflox-pkgdb.dylib' (no such file), 'lib/libflox-pkgdb.dylib' (no such file), '/usr/local/lib/libflox-pkgdb.dylib' (no such file), '/usr/lib/libflox-pkgdb.dylib' (no such file, not in dyld cache)
```

Reply:
This should be fixed.
Thanks to @bryanhonof for helping get a Darwin box, and @ghudgins for patiently helping sanity check.